extensions/aris/src/bigDataCluster/controller/controller.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { EndpointRouterApi } from './apiGenerated';
+import { IEndPointsResponse, IControllerError, IEndPointsRequest, IHttpResponse, IEndPoint } from './types';
+
+export class ArisController {
+
+	private static adjustUrl(url: string): string {
+		if (!url) {
+			return undefined;
+		}
+
+		url = url.trim().replace(/ /g, '').replace(/,(\d+)$/, ':$1');
+		if (!url.includes('://')) {
+			url = `https://${url}`;
+		}
+		return url;
+	}
+
+	public static async getEndPoints(
+		url: string, username: string, password: string, ignoreSslVerification?: boolean
+	): Promise<IEndPointsResponse> {
```

Comment:
IEndpointsResponse is a very detailed thing to return to a public caller. Consider returning a simplified response - what does the dialog actually need to know here?

Reply:
I temporarily commented out the fields not being used for now. There will be server api change, and some field might become necessary.